lib/internal/vm/module.js

```diff
@@ -252,8 +253,43 @@ function validateInteger(prop, propName) {
   }
 }
 
+async function getWrapFromModule(m, scriptOrModule, linker) {
+  if (!m || !wrapMap.has(m))
+    throw new ERR_VM_MODULE_NOT_MODULE();
+
+  if (scriptOrModule instanceof Module &&
+      (m.context !== scriptOrModule.context))
+    throw new ERR_VM_MODULE_DIFFERENT_CONTEXT();
+
+  const childLinkingStatus = linkingStatusMap.get(m);
+
+  if (childLinkingStatus === 'errored')
+    throw new ERR_VM_MODULE_LINKING_ERRORED();
+  if (childLinkingStatus === 'unlinked')
+    await m.link(linker);
+
+  return wrapMap.get(m);
+}
+
+async function callLinkerForModuleWrap(linker, specifier, scriptOrModule) {
+  const m = await linker(specifier, scriptOrModule);
+  return getWrapFromModule(m, scriptOrModule, linker);
+}
+
+async function callLinkerForNamespace(linker, specifier, scriptOrModule) {
```

Comment:
What happens if returning a vm.Module instance that has not been linked (and that has dependencies)? Can we have a test for this error case?

Reply:
this is already tested by `testScript`